chore: Improve 4 projects readme and troubleshooting (#998)
terraform-google-modules/terraform-example-foundation: docs/TROUBLESHOOTING.md

This is a transient error and the deploy can be retried. Wait for at least a minute and retry the deploy.

### Error: Unknown project id on 4-project step context

**Error message:**

```text
Error 400: Unknown project id: 'prj-<business-unity>-<environment>-sample-base-<random-suffix>', invalid
```

**Cause:**

When you try to run 4-projects step without requesting additional project quota for **project service account created in 0-bootstrap step** you may face the error above, even after the project quota issue is resolved, due to an inconsistency in terraform state.

**Solution:**

- Make sure you [have requested the additional project quota](#project-quota-exceeded) for the **project SA e-mail** before the following steps.

You will need to mark some resources as tainted in order to trigger the proper missing project's recreation to fix the inconsistent terraform state by following the next steps.

> 1. In your local, go to the path that the error is being reported.

Reviewer: ```suggestion
1. In a terminal, navigate to the path where the error is being reported.
```

>    For example, if the unknown project ID is `prj-bu1-p-sample-base-abcd`, you should go to ./gcp-projects/business_unit_1/production (business_unit_1 due to `bu1` and production due to `p`).

Reviewer: Linking to the docs/or white paper for the naming conventions may be worthwhile here if there is a suitable link.

Reviewer 2: fixed

   ```bash
   cd ./gcp-projects/<business_unit>/<environment>
   ```

> 1. Run terraform init command so you can pull the remote state to your local.

Reviewer: ```suggestion
1. Run terraform init command so you can pull the remote state.
```

   ```bash
   terraform init
   ```
1. Run terraform state list command, filtering by `random_project_id_suffix`. This command will give you all the expected projects that should be created for this BU and environment that uses a random suffix.

   ```bash
   terraform state list | grep random_project_id_suffix
   ```

> 1. Run gcloud projects list command. You should replace `id_of_the_environment_folder` with the proper ID of the folder. This command will give you all the projects that were actually created.

Reviewer: Where does the user get the folder ID from? The previous step? can you explain how

Reviewer 2: fixed

   ```bash
   gcloud projects list --filter="parent=<id_of_the_environment_folder>"
   ```
1. For each resource listed in terraform state step that's **not** returned by gcloud projects list step, we should mark as tainted to be recreated in order to fix the inconsistency in terraform state.

   ```bash
   terraform taint <resource>[index]
   ```

>    For example, in the following command we are marking as tainted the env secrets project. You may need to run the taint command multiple times, depending on how many missing projects you do have.

Reviewer: ```suggestion
   For example, in the following command we are marking as tainted the env secrets project. You may need to run the taint command multiple times, depending on how many missing projects you have.
```

   ```bash
   terraform taint module.env.module.env_secrets_project.module.project.module.project-factory.random_string.random_project_id_suffix[0]
   ```

> 1. Finally, after run the taint command for all the non-matching items, you can go to Cloud Build and trigger the retry action for the failed job. You may have a success build or some similar error for another BU/environment that will require the same solution, but changing the proper paths according to the BU/environment reported in the error log.

Reviewer: ```suggestion
1. Finally, after running the taint command for all the non-matching items, you can go to Cloud Build and trigger a retry action for the failed job. This should complete successfully, if you encounter another some similar error for another BU/environment that will require you to follow this guide again but instead changing paths according to the BU/environment reported in the error log.
```
